lib/parser/tokens.js

```diff
@@ -27,6 +27,11 @@ const HASH = createToken({
 , pattern: /#/
 })
 
+const THIS = createToken({
+  name: 'this'
+, pattern: /this/
```

Comment:
Shouldn't it have a word boundaries? Wouldn't something like thisthing match?

Reply:
No this is how it wants you to define token patterns generally. adding boundaries just treats it as an `identifier`

Reply:
![image](https://user-images.githubusercontent.com/148561/184544220-f7ac8c29-1fb3-4e18-be26-09763e06c6bb.png)